libvast/src/view.cpp

```diff
@@ -41,30 +43,54 @@ bool operator==(subnet_view x, subnet_view y) noexcept {
   return x.network_ == y.network_ && x.length_ == y.length_;
 }
 
-vector_view::iterator vector_view::begin() const {
-  // TODO: const_cast okay?
-  return {vector_view_ptr{const_cast<vector_view*>(this), true}, 0};
+
+default_vector_view::default_vector_view(const vector& xs) : xs_{xs} {
+  // nop
+}
+
+default_vector_view::value_type default_vector_view::at(size_type i) const {
+  return make_data_view(xs_[i]);
 }
 
-vector_view::iterator vector_view::end() const {
-  // TODO: see above
-  return {vector_view_ptr{const_cast<vector_view*>(this), true}, size()};
+default_vector_view::size_type default_vector_view::size() const noexcept {
+  return xs_.size();
 }
 
-default_vector_view::default_vector_view(const vector& xs) : xs_{xs} {
+
+default_set_view::default_set_view(const set& xs) : xs_{xs} {
   // nop
 }
 
-vector_view::value_type default_vector_view::at(size_type i) const {
-  return make_data_view(xs_[i]);
+default_set_view::value_type default_set_view::at(size_type i) const {
+  auto it = xs_.begin();
```

Comment:
I'm confused by the use of iterators here. Why not `return make_data_view(xs_[i])`?

In case there's no way around iterators then these three lines can get shortened to: `return make_data_view(*std::next(xs_.begin(), i))`.

Or even shorter for random-access iterator types (such as `std::vector<T>::iterator`): `return make_data_view(xs_.begin()[i])` .

Reply:
> I'm confused by the use of iterators here. Why not return make_data_view(xs_[i])?

Currently, `vast::set` is an alias over `std::set<data>`, which does not have a positional `operator[]`.

I'll wind up using this: `return make_data_view(*std::next(xs_.begin(), i))`